### DISCUSSION TITLE
Deleting document is not deleting it from SQL table

### DISCUSSION THREAD

**Discussion by nickchampion** [ravendb/ravendb#16868] [General]
Hi,

I have an SQL ETL ongoing task set up which is working as expected except when it comes to deletes, when I physically delete a document from RavenDB there is no delete statement being sent to my SQL database so I'm ending up with rows which should not be there.

Is there anything I need to do in the ETL setup to ensure deletes propagate?

Version cloud 5.4.107

I have noticed the following alert, not sure if its related

<img width="808" alt="image" src="https://github.com/ravendb/ravendb/assets/3949005/13a88b12-8ca0-4772-898a-55194abab4ab">

**Comment by ayende** [comment]
It is probably not related.
Question, do you have defined: `Insert Only` in the ETL?

**Reply by nickchampion** [reply]
Hi Oren, no none of our tables have the insert only flag enabled

**Reply by ayende** [reply]
Can you try to reproduce this with the debug logs enables?

**Reply by nickchampion** [reply]
Attached are admin logs, I have triggered an update to a document (which writes to 3 tables) the update is working as expected the name is successfully changed in the SQL table after editing the doc in RavenDB.

The delete however is not working, I can see the delete statements generated in the logs and if I execute these on the SQL database they work, there are no foreign constraints which could prevent deletion

[admin-log-2023-07-18 12-12.txt](https://github.com/ravendb/ravendb/files/12079739/admin-log-2023-07-18.12-12.txt)

Thanks

**Reply by ayende** [reply]
Okay, that is helpful.
I assume that you don't have any `instead of` triggers there?
If you run those queries manually, do they work?

As far as I can see, we are doing everything properly. If you use SQL Profiler, do you see the behavior?

**Reply by nickchampion** [reply]
Hi @ayende 

Just got back to looking at this again. I have enabled logging on the database and can see the issue. RavenDB is issuing a delete statement but lowercasing the id's as follows

```
DELETE FROM "harvest_plan_loads" WHERE "id" IN ('harvestplanloads/708-a')
```

This is not matching any records, if the id is in the correct case it works

```
DELETE FROM "harvest_plan_loads" WHERE "id" IN ('HarvestPlanLoads/708-A')
```

This only seems to be the case when a document is deleted, when its updated the ID is in the correct casing and documents are deleted and re-inserted successfully.

I'm using Postgres 14 on AWS Aurora

Any recommendations as to how best to deal with this?

**Reply by ayende** [reply]
Can you change the `id` collation on the destination database?
The issue there is that RavenDB doesn't keep track of the casing for tombstones of deletes documents.

**Reply by nickchampion** [reply]
Thanks @ayende I've changed the collation on ID columns which has resolved the issue.

It might be worth adding this to the SQL ETL docs as it will no doubt catch others with case sensitive collation settings.

Thanks

Nick

**Reply by arekpalinski** [reply]
Thanks. We'll update the docs. CC @reebhub

**Comment by reebhub** [comment]
thx
[RDoc-2506](https://issues.hibernatingrhinos.com/issue/RDoc-2506/Document-SQL-ETL-issue-with-SQL-column-collation-when-deleting-a-record)
